src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -176,6 +178,42 @@ public static boolean itinerariesMatch(Itinerary referenceItinerary, Itinerary c
         return true;
     }
 
+    /**
+     * Checks that the specified itinerary is on the same day as the specified date/time.
+     * @param itinerary the itinerary to check.
+     * @param requestDateTime the request date/time to check, in the OTP's time zone.
+     * @param tripIsArriveBy true to check the itinerary endtime, false to check the startTime.
+     * @return true if the itinerary's startTime or endTime is one the same day as the day of the specified date and time.
+     */
+    public static boolean isSameDay(Itinerary itinerary, ZonedDateTime requestDateTime, boolean tripIsArriveBy) {
+        // TODO: Make SERVICE_DAY_START_HOUR an optional config parameter.
+        final int SERVICE_DAY_START_HOUR = 3;
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(DEFAULT_DATE_FORMAT_PATTERN);
```

Comment:
We construct default formatters in a couple of places. Opportunity to store a formatter on/import from DateTimeUtils instead?

Reply:
Removed in #112.